airflow/serialization/serde.py

```diff
@@ -360,7 +360,7 @@ def _register():
 @functools.lru_cache(maxsize=None)
 def _get_patterns() -> list[Pattern]:
     patterns = conf.get("core", "allowed_deserialization_classes").split()
-    return [re2.compile(re2.sub(r"(\w)\.", r"\1\..", p)) for p in patterns]
+    return [re2.compile(re2.sub(r"(\w)\.?\*", r"\1\.", p)) for p in patterns]
```

Comment:
I still find it quite a bit mind boggling and canntot sey that it is wrong or right. If we want to keep the flexibility there. I think we should split it into two different checks.

1) Check assuming that we are using glob

If not matched:

2) Another check assuming that we are using regexp.

This would be much easier to reason about it an we could also find a reasonable way how to explain it to users - because currently it's next to impossible to explain this behaviour to the user. 

As the old saying is "you have problems, introduce regexp - now you have two problems". In this case I feeel with two regexpe we exponentially grow complexity of what happens here.

I think we could explain to the users easily that you can either use glob or regexp and both will match. There is a small risk of matching something that we did not expect but I think it is very unlikely.

we could also add a config or prefix to make it more deterministic (say `deserialization_classes_match`: [regexp, glob, both]` with both being default.

@bolkedebruin  WDYT ? I find it very difficult to reason and confirm if the proposed regexp works as expected.

Reply:
Hey @potiuk, thanks for the feedback! The decision to fix the problem only by changing the regexp already used was solely for simplifying the solution. 
However I agree that a separated approach (splitting the cases) would be easier for future devs to understand whats going, as well as, as to test the functions. Thus, if the explanation to how the regexp works is not clear, we're happy to make these changes! 
The only thing that may worry me is the idea to add a config to specify which type, which may just add a layer of complexity that might not be needed, as it looks like it would be very unlikely for something to be matched unexpectedly if done and tested well enough.

Reply:
To answer / give an opinion on the easiest part: I don't think a config option is a good idea for the same reasons that @tobiaszorzetto mentioned - including the cognitive load on people that do not know what a `glob` is etc. 

Although alternatively a `allowed_serialization_classes` could be the glob version and `allowed_serialization_classes_regexp` could be the regexp version. Could that work?

Either way I do think a more formal documentation inside the function to explain what it tries to do and also having the tests more explicitly stating at each test what they are testing would make sense.

Reply:
Sure, will do that!

Reply:
And we can really treat it as a bugfix, I think the way it was defined from the very beginning was a bug on its own :)

Reply:
Just one think - make sure to add a `significant` newstfragment for it (see `newsfragments` folder) - this change will likely break someone's workflows, so we need to give the users some hint on how they shoudl fix the problems they might have (if they do). I think it's rather unlikely it will affect someone's deployment, so that's why bugfix, but we should have answer ready for users if it does @tobiaszorzetto

Reply:
Just added the changes we discussed in this convo. @potiuk @bolkedebruin if you guys could let me know what you think and if you have any other suggestions, that would be awesome. :)